<?xml version="1.0"?>
<robot name="delto_5f" xmlns:xacro="http://www.ros.org/wiki/xacro">


    
    <xacro:property name="mesh_path" value="$(find delto_description)/meshes/DG5F"/>
    
    <!-- 기존 URDF include -->
    <xacro:include filename="$(find delto_description)/urdf/delto_gripper_5f.urdf" />

    <!-- base_link과 dummy_link는 유지 (충돌 없음) -->
    <link name="base_link" />

    <joint name="base_joint" type="fixed">
        <parent link="base_link"/>
        <child link="dummy_link"/>
        <origin xyz="0 0 0" rpy="0 0 0"/>
    </joint>


    <ros2_control name="GazeboSimSystem" type="system">
        <hardware>
            <plugin>ign_ros2_control/IgnitionSystem</plugin>
        </hardware>
        
        <!-- Finger 1 joints -->
        <joint name="j_1_1">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_1_2">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_1_3">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_1_4">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <!-- Finger 2 joints -->
        <joint name="j_2_1">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_2_2">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_2_3">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_2_4">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <!-- Finger 3 joints -->
        <joint name="j_3_1">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>  
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_3_2">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_3_3">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_3_4">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <!-- Finger 4 joints -->
        <joint name="j_4_1">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_4_2">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_4_3">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_4_4">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <!-- Finger 5 joints -->
        <joint name="j_5_1">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_5_2">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_5_3">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>

        <joint name="j_5_4">
            <command_interface name="position">
                <param name="min_position">-0.5</param>
                <param name="max_position">0.5</param>
            </command_interface>
            <state_interface name="position">
                <param name="initial_position">0.0</param>
            </state_interface>
            <state_interface name="velocity"/>
            <state_interface name="effort"/>
        </joint>
    </ros2_control>

    <gazebo>
    <self_collide>false</self_collide>
        <!-- <plugin filename="libign_ros2_control-system.so" name="ign_ros2_control::IgnitionROS2ControlPlugin">
            <robot_param>robot_description</robot_param>
            <robot_param_node>robot_state_publisher</robot_param_node>
            <parameters>$(find delto_5f_driver)/cfg/gazebo_controller.yaml</parameters>
        </plugin> -->

        <plugin name="ign_ros2_control::IgnitionROS2ControlPlugin" filename="libign_ros2_control-system.so">
  <robot_param_node>robot_state_publisher</robot_param_node>
  <robot_param>robot_description</robot_param>
  <parameters>$(find delto_5f_driver)/cfg/gazebo_controller.yaml</parameters>
  <ros>
            <namespace>/</namespace>
        </ros>
</plugin>

    </gazebo>

</robot>